Yearn-agents/InstantWithdrawl/src/blockchain.ts

```diff
@@ -0,0 +1,32 @@
+import path from "path";
+import dotenv from "dotenv";
+const ganache = require("ganache-cli");
+dotenv.config();
+
+const FORK_TESTS_CACHE_PATH = path.join(
+  __dirname,
+  "..",
+  "provider",
+  ".hardhat_node_test_cache"
+);
+
+export const DEFAULT_HARDFORK = "byzantium";
+
+export default function runServer(accounts: Array<string>) {
+  const options = {
+    chainId: 1,
+    networkId: 1,
+    port: 9545,
+    fork: process.env.jsonRPC,
+    hardfork: DEFAULT_HARDFORK,
+    fork_block_number: 12628504,
```

Comment:
This should be the block number where you are checking.

Reply:
Not really.  https://ethereum.stackexchange.com/questions/90194/truffle-ganache-fork . The number specifies that the chain will fork from this blocknumber, but all the other new blocks are also included i.e. anything before this block is not accessbile, but everything after this is. And the web3 call is made to the latest block on the chain.

Reply:
Not really, forking from block `X` means that you will be adding blocks on top of block `X`. You can do a simple test for checking that.